components/badge/badge_input.c

```diff
@@ -8,6 +8,7 @@
 #include <esp_log.h>
 
 #include "badge_input.h"
+#include "badge_vibrator.h"
```

Comment:
make sure the badge_vibrator is initialized when used.
I think this can be done best with an 'esp_err_t badge_input_enable_haptic_feedback()' method which initializes the badge_vibrator module. (as added bonus, the haptic feedback can be enabled/disabled dynamically)